presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,228 @@
+        }
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                return jdbcTypeToPrestoType(typeHandle);
+            default:
+                return super.toPrestoType(session, connection, typeHandle);
+        }
+    }
+
+    private Optional<ColumnMapping> jdbcTypeToPrestoType(JdbcTypeHandle type)
+    {
+        int columnSize = type.getColumnSize();
+        if (columnSize > VarcharType.MAX_LENGTH || columnSize == -1) {
+            return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+        }
+        return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session))) {
+            try (ResultSet resultSet = getDruidColumns(tableHandle, connection.getMetaData())) {
+                List<JdbcColumnHandle> columns = new ArrayList<>();
+                while (resultSet.next()) {
+                    JdbcTypeHandle typeHandle = new JdbcTypeHandle(
+                            resultSet.getInt("DATA_TYPE"),
+                            // type_name isn't supported in Druid
+                            Optional.of(resultSet.getString("TYPE_NAME")),
+                            resultSet.getInt("COLUMN_SIZE"),
+                            resultSet.getInt("DECIMAL_DIGITS"),
+                            // array_dimensions isn't supported in Druid
+                            Optional.ofNullable(null));
+                    Optional<ColumnMapping> columnMapping = toPrestoType(session, connection, typeHandle);
+                    // skip unsupported column types
+                    if (columnMapping.isPresent()) {
+                        String columnName = resultSet.getString("COLUMN_NAME");
+                        boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
+                        columns.add(new JdbcColumnHandle(columnName, typeHandle, columnMapping.get().getType(), nullable));
+                    }
+                }
+                if (columns.isEmpty()) {
+                    // In rare cases (e.g. PostgreSQL) a table might have no columns.
+                    throw new TableNotFoundException(tableHandle.getSchemaTableName());
+                }
+                return ImmutableList.copyOf(columns);
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    /**
+     * Custom way of retrieving column names since the {@link BaseJdbcClient#getColumns(JdbcTableHandle, DatabaseMetaData)}
+     * method uses character escaping that doesn't jive well with Druid SQL.
```

Comment:
Then it might be not correct -- when asking for `some_table`, you should also get columns from `somertable`, (unless Druid JDBC not only does not support escaping, but also does not support patterns).

If I am right, and there really is no way to escape, we need to filter on our side as well.

Reply:
One of the problems I found with ```BaseJdbcClient#getColumns(JdbcTableHandle, DatabaseMetaData)``` is how it handles `_`.  

passing ```orders_1``` to ```escapeNamePattern("orders_1", "\")``` changes the string to 
```orders\_1```. And so when the call is made to Druid to get the metadata using:

```
protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData metadata)
            throws SQLException
    {
        return metadata.getColumns(
                tableHandle.getCatalogName(),
                escapeNamePattern(Optional.ofNullable(tableHandle.getSchemaName()), metadata.getSearchStringEscape()).orElse(null),
                escapeNamePattern(Optional.ofNullable(tableHandle.getTableName()), metadata.getSearchStringEscape()).orElse(null),
                null);
    }
```
the call fails since Druid doesn't remove the escape character `\`. So I am not sure if escaping is going to work here. 

Druid though, by itself, does support patterns. 
See: 
https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/avatica/DruidMeta.java#L457

Reply:
That's strange, because it reports `\` in `java.sql.DatabaseMetaData#getSearchStringEscape`...

If we cannot use search escapes, this would mean we need to add filtering on our side. 
You can test this with:
- create `some_table` in druid, with `a` column
- create `somertable` in druid, with `b` column
- run `DESCRIBE some_table` 

i expect that, with current code, it will report both `a` and `b` columns.